File: 0443-string-compression/0443-string-compression.js
/**
 * @param {character[]} chars
 * @return {number}
 */
var compress = function(chars) {
  let str = "";
  let count = 1;

  for (let i = 0; i < chars.length; i++) {
    if (chars[i] === chars[i + 1]) {
      count++;
    } else {
      if (count === 1) {
        str += chars[i];
      } else {
        str += (chars[i] + count);
      }

      count = 1;
    }
  }
  
  const result = str.split("");
  
  chars.splice(0, chars.length, ...result);

  return result.length;
};

File: 0443-string-compression/0443-string-compression.py
from typing import List

class Solution:
    def compress(self, chars: List[str]) -> int:
        result = ""
        count = 1

        for i in range(0, len(chars) - 1):
            if chars[i] == chars[i + 1]:
                count += 1
            else:
                if count > 1:
                    result += chars[i] + str(count)
                else:
                    result += chars[i]
                
                count = 1

        if count > 1:
            result += chars[-1] + str(count)
        else:
            result += chars[-1]

        index = 0

        for char in result:
            chars[index] = char
            index += 1
            
        return len(result)
